tests/test_emissions_tracker.py

```diff
@@ -38,11 +42,20 @@ class TestCarbonTracker(unittest.TestCase):
     def setUp(self) -> None:
         self.data_source = get_test_data_source()
         self.project_name = "project_foo"
-        self.emissions_file_path = os.path.join(os.getcwd(), "emissions.csv")
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.temp_path = Path(self.temp_dir.name)
+        self.emissions_file_path = self.temp_path / "emissions.csv"
+        # builtins.open is patched not to open ~/.codecarbon.config nor
+        # ./.codecarbon.config so that the user's local configuration does not
+        # alter tests
+        patcher = mock.patch(
+            "builtins.open", new_callable=get_custom_mock_open(empty_conf, empty_conf)
+        )
+        self.addCleanup(patcher.stop)
+        patcher.start()
```

Comment:
You likely wants to clean the patcher in `tearDown` as well

Reply:
I thought `addCleanup` would do that